Sources/ErgastAPI/Endpoint.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  Endpoint.swift
+//  
+//
+//  Created by Giovanni Noa on 4/28/20.
+//
+
+import Foundation
+
+/// Object to use for generating a URL with a given Path enum case..
+struct Endpoint {
+    private var path = "/api/f1"
+    
+    
+    /// Initializer for an Endpoint object.
+    /// - Parameters:
+    ///   - path: Specify a path, mapping to a specific endpoint of the Ergast REST API.
+    ///   - season: Specify either all historical circuits, or circuits for a specific year.
+    init(with path: Path, for season: Season?) {
+        if let season = season {
+            self.path.append(path.subPath(for: season))
```

Comment:
This took me a few reads to figure out what was happening. 
The result of the `Path` enum's `subPath(for:)` method is appended to the `path` var. That's quite the mouthful.

Assuming the `Endpoint.path` string isn't going to change, then what if it were a private constant instead and maybe rename it to `urlPath`. Then in the `Path` enum we add a new helper which can return the full URL path since its in charge of all path-related things.

So here in the `Endpoint` constructor:

```swift
self.urlPath = path.urlPath(for: season)
```
And within the `Path` enum, we make the `/api/f1` string be a private constant that the helper function uses. Something like:
```swift
private let basePath = "/api/f1"

func urlPath(for season: Season?) -> String {
  guard let subPath = subPath(for season: Season) else {
    return basePath
  } 
   return basePath + subPath
}
```